src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1110,11 +1110,40 @@ bool entitiesQuery
     }
 
     /* Process query result */
-    while (cursor->more()) {
+    while (cursor->more())
+    {
+        BSONObj                 r    = cursor->next();
+        std::string             err  = r.getStringField("$err");
+        ContextElementResponse* cer  = new ContextElementResponse();
+
+        if (err != "")
+        {
+          //
+          // We can't return the error 'as is', as it may contain forbidden characters.
+          // So, we can just match the error and sent a less descriptive text
+          //
+          const char* invalidPolygon      = "Exterior shell of polygon is invalid";
+          const char* defaultErrorString  = "Internal mongo problem";
+
+          LM_W(("Database Error (%s)", err.c_str()));
+
+          if (strncmp(err.c_str(), invalidPolygon, strlen(invalidPolygon)) == 0)
+            err = invalidPolygon + std::string(" - not detected at creation time");
+          else
+            err = defaultErrorString;
+
+          //
+          // FIXME P1: It would be nice to fill in the entity but it is
+          //           difficult to do this, I simply fill it in with enV[0] ...
+          //
+          cer->contextElement.entityId.fill(enV[0]);
```

Comment:
In this case, the problem in a general case is not actually with one entity or the other, but with the query send to mongo, that has "mixed" information of all the entities in the request. Thus, I only see two possible options:

1. Return all the entities
2. Don't return any entity (NGSI response format allows this?)

In my opinion 2 is better (simpler code and shorter response to the client). There is no loss of information, as the client already knows the entities that it included in the requesz producing that response.